### PR TITLE
PR to fix dropdown and max button bugs with addition of max button to addFundsModal

### PR DESCRIPTION
**Pull request by souvikmishra**
This PR was initially opened to just fix drop-down and max button issues in withdraw fund modal (hence the PR title) but since then the scope of it has changed quite a bit. This PR now does the following things:
- fix the drop-down (i.e. `<Select />` component) misalignment issue in modals (due to DaisyUI's internal change) by positioning each drop-down absolutely w.r.t to their parent collapse button.
- in doing the above change we removed all overflow properties, this posed a challenge wherein if the content in a modal exceeds modal height then the content overflow breaks the UI, to cater to this we now pass a `isScrollable` attribute to Modal such that we have some control over when to add the scroll behavior back.
- added alternate styles to the `<ChainSwitcher />` component so that it can be used in tandem with `<NetworkPrompt />` and updated its implementation to use the `<details><details/>` tag.
- fixed bandwidth calculation in the `<JobDetailsModal />` to cater to up-scaled bandwidth rates.
- fixed the max button bug where in the value being populated had commas which by convention is an invalid character for calculations.
- added `<MaxButton />` feature to `<AddFundsModal />`
- refactored `OysterConstants.ts`, `OysterStore.ts` and other affected files to drive all oysterToken based calculations through chainConfigs.
- renamed all scaled values by adding a suffix 'Scaled' for better readability.

### CODE REVIEW

File: src/lib/page-components/oyster/sub-components/AddFundsToJob.svelte [hunk 1]
```diff
@@ -135,14 +119,16 @@
...
 	$: instanceCost = computeCost(duration || 0, instanceRate);
-	$: invalidCost = !instanceCost || !(maxBalance >= instanceCost / OYSTER_RATE_SCALING_FACTOR);
+	$: invalidCost =
+		!instanceCost || !($walletBalanceStore.pond >= instanceCost / OYSTER_RATE_SCALING_FACTOR);
 	$: inValidMessage = !instanceCost
 		? ''
 		: invalidCost
 		? `Insufficient balance. Your current wallet has ${bigNumberToString(
-				maxBalance,
+				$walletBalanceStore.pond,
```
Comment (prateekreddy) [inline src/lib/page-components/oyster/sub-components/AddFundsToJob.svelte:131]: this should be USDC balance instead right ?

File: src/lib/utils/helpers/oysterHelpers.ts [hunk 2]
```diff
@@ -505,12 +510,10 @@ export const addRegionNameToMarketplaceData = (
 export function getBandwidthFromRateAndRegion(bandwidthRate: bigint, region: string) {
 	const rateForRegion = getBandwidthRateForRegion(region);
 	if (rateForRegion === undefined) return 0n;
+	// this is done to ceil the number since rateForRegion
+	// is essentially, actualRate * OYSTER_RATE_SCALING_FACTOR
 	const bandwidthWithAllPrecision =
```
Comment (prateekreddy) [inline src/lib/utils/helpers/oysterHelpers.ts:515]: Let's maintain a common terminology for all places where the values are scaled

File: src/lib/page-components/oyster/sub-components/AddFundsToJob.svelte (hunk 1 above)
Comment (souvikmishra) [reply src/lib/page-components/oyster/sub-components/AddFundsToJob.svelte:131]: yea it should be the currency that we set in the chain as the oyster_token

File: src/lib/chains/arbGoerli.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ export const ARB_GOERLI: ChainConfig = {
 		OYSTER: '0xdbc4779b69bdc0f657ba35648e434bae042a00e0',
 		POND: '0xa9472a9C135Cb33221de441a4DEd393FD515a69C',
 		MPOND: '0x0B3d9b224496C2A2Fa1a4096D8EB4D350eFd9079',
-		USDC: '0x53F3Cacb08F1c3ee7dC4fac2d1B73482e5d56423'
+		USDC: '0x8FB1E3fC51F3b789dED7557E680551d93Ea9d892'
```
[the remark this answers is not in the document]
Comment (souvikmishra) [reply src/lib/chains/arbGoerli.ts:19]: Testnet USDC that I bridged pointed to this address

File: src/lib/chains/arbGoerli.ts [hunk 4]
```diff
@@ -34,21 +34,29 @@ export const ARB_GOERLI: ChainConfig = {
...
+			symbol: '$'
 		},
 		MPOND: {
-			token_decimals: 18,
-			token_precision: 6,
-			symbol: 'MPOND'
+			decimal: 18,
+			precision: 6,
+			currency: 'MPOND',
+			symbol: '$'
```
[the remark this answers is not in the document]
Comment (souvikmishra) [reply src/lib/chains/arbGoerli.ts:53]: fixed

File: src/lib/chains/arbGoerli.ts [hunk 5]
```diff
@@ -34,21 +34,29 @@ export const ARB_GOERLI: ChainConfig = {
...
+		rateReviseWaitingTime: 5 * 60 // 5 minutes
+	},
 	tokens: {
 		POND: {
-			token_decimals: 18,
-			token_precision: 2,
-			symbol: 'POND'
+			decimal: 18,
+			precision: 2,
+			currency: 'POND',
```
[the remark this answers is not in the document]
Comment (souvikmishra) [reply src/lib/chains/arbGoerli.ts:46]: we use this token metadata inside a store, so either we can maintain them here or create that field on store level

**Review by prateekreddy — APPROVED** [review]

**Pull request merged**